### PR TITLE
Faster Control Polling

### PR DESCRIPTION
**Pull request by zoul0813**

### CODE REVIEW

File: examples/snake/src/snake.c [hunk 1]
```diff
@@ -152,10 +162,14 @@ static void draw_background(void) {
...
 
-static void update_score(void) {
-    char text[3];
-    increment_score(snake.score, text);
-    nprint_string(text, 3, WIDTH - 4, HEIGHT);
+static void update_stat(void) {
+    char text[7];
+    sprintf(text,"SP:%02d", snake.speed);
+    nprint_string(text, strlen(text), 1, HEIGHT);
+    sprintf(text,"B:%02d", snake.apples_to_boost);
```
Comment (Zeal8bit) [inline examples/snake/src/snake.c:169]: It is hard to guess what this means when playing (and not seeing the code), maybe we can get rid of the `speed` meter and only keep this `next` level/step info?

File: examples/snake/src/snake.h [hunk 2]
```diff
@@ -59,8 +59,10 @@ typedef struct {
     uint8_t former_direction;
     uint8_t just_ate;
     /* BCD representation of the score */
-    uint8_t score[2];
-    /* 0: slowest, 100: fastest */
+    // uint8_t score[2];
```
Comment (Zeal8bit) [inline examples/snake/src/snake.h:62]: You can get rid of this comment now and remove the assembly score related to it

**Comment by Zeal8bit** [comment]
Thanks a lot for your contribution, before merging, please squash your commits

File: examples/snake/src/snake.c [hunk 3]
```diff
@@ -56,7 +58,7 @@ const uint8_t letters_palette[] = {
   0x00, 0x00, 0x83, 0xa8, 0xcb, 0xf1, 0xc7, 0xfc, 0x52, 0xff, 0xff, 0xff
 };
 
-int main(int argc, char** argv) {
+uint8_t main(int argc, char** argv) {
```
Comment (Zeal8bit) [inline examples/snake/src/snake.c:61]: This is part of Zeal 8-bit OS C API, it shall not be changed (SDCC doesn't put the return values in the same registers if the type is different)

**Comment by Zeal8bit** [comment]
Thanks a lot, merging!

File: examples/snake/src/snake.c (hunk 3 above)
Comment (zoul0813) [reply examples/snake/src/snake.c:61]: Ah - ok, my bad

**Pull request merged**